### PR TITLE
MF-687: Extend desktop search overlay to cover left navigation

### PR DESCRIPTION
**Pull request by donaldkibet**
### Description

1. Extended desktop search to cover the left navigation
2. Updated the search experience to match the designs as shown [here](https://app.zeplin.io/project/60d59321e8100b0324762e05/dashboard?tag=Patient%20search)

### Desktop Search Experience
desktop

### Tablet search experience
tablet

### CODE REVIEW

File: packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx [hunk 1]
```diff
@@ -1,47 +1,163 @@
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useCallback, useState, useEffect, useReducer, useMemo } from 'react';
 import Search20 from '@carbon/icons-react/es/search/20';
+import Close20 from '@carbon/icons-react/es/close/20';
 import { HeaderGlobalAction } from 'carbon-components-react/es/components/UIShell';
 import styles from './patient-search-icon.component.scss';
 import PatientSearch from '../patient-search/patient-search.component';
+import Search from 'carbon-components-react/es/components/Search';
+import { useTranslation } from 'react-i18next';
+import { Button } from 'carbon-components-react';
```
Comment (FlorianRappl) [inline packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx:9]: Use submodule import.

File: packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx [hunk 2]
```diff
@@ -1,47 +1,163 @@
...
+            index: i + 1,
+          }));
+          dispatch({ type: ActionTypes.resolved, payload: results });
+        },
+        (error) => {
+          dispatch({ type: ActionTypes.error, payload: error });
+        },
+      );
+    }
+    return () => ac.abort();
```
Comment (FlorianRappl) [inline packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx:95]: Why is a function returned here? Do you intend to use this in a side-effect hook? Right now its not.

File: packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx [hunk 3]
```diff
@@ -1,47 +1,163 @@
...
+  const { t } = useTranslation();
+  const layout = useLayoutType();
   const [open, setOpen] = useState<boolean>(false);
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const initialState: PatientSearch = { status: 'idle', searchResults: [] };
+  const [{ searchResults, status }, dispatch] = useReducer(reducer, initialState);
 
-  const togglePatientSearch = useCallback(() => {
-    setOpen((prevState) => !prevState);
+  const performSearch = () => {
```
Comment (FlorianRappl) [inline packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx:78]: The whole callback here screams for being used inside a side effect hook.

In the current state

- cancellation will never work
- if the component is unmounted during the resource loading we get state errors left and right

**Review by FlorianRappl — CHANGES_REQUESTED** [review]

**Review by FlorianRappl — APPROVED** [review]
LGTM!

**Comment by gracepotma** [comment]
THANK YOU FOR THIS ONE DONALD! Very nice to have this done. It was bugging me 😅

**Pull request merged**